library/swift/HeadersContainer.swift

```diff
@@ -0,0 +1,110 @@
+/// The container which manages the underlying headers map.
+/// It maintains the original casing of passed header names.
+/// It treats headers names as case-insensitive for the purpose
+/// of header lookups and header name conflict resolutions.
+struct HeadersContainer: Equatable {
+  private var headers: [String: Header]
+
+  // Represents a headers name together with all of its values.
+  // It preserves the original casing of the header name.
```

Comment:
Nit: this reads as a doctring

```suggestion
  /// Represents a headers name together with all of its values.
  /// It preserves the original casing of the header name.
```